src/ocean/util/log/Stats.d

```diff
@@ -650,3 +656,369 @@ unittest
         }
     }
 }
+
+
+/**************************************************************************
+
+    Struct that parses a stats line
+
+**************************************************************************/
+
+public struct StatsLine
+{
+    /**************************************************************************
+
+        The stats after the date and time were removed
+
+    **************************************************************************/
+
+    private Buffer!(char) line;
+
+    /**************************************************************************
+
+        The line date
+
+    **************************************************************************/
+
+    public cstring date;
+
+    /**************************************************************************
+
+        The line time
+
+    **************************************************************************/
+
+    public cstring time;
+
+    /**************************************************************************
+
+        Create an StatsLine
+
+        Params:
+            line = a line stat
+
+        Returns:
+            A StatsLine
+
+    **************************************************************************/
+
+    static public StatsLine opCall (cstring line)
+    {
+        static Buffer!(char) buffer_line;
+
+        StatsLine stats_line;
+        buffer_line = line;
+        auto len = buffer_line.length;
+
+        auto date_end_position = find(buffer_line, " ");
+        stats_line.date = buffer_line[0 .. date_end_position];
+
+        auto time_begin_position = date_end_position + 1;
+        auto time_end_position = find(buffer_line[time_begin_position .. len], " ") + time_begin_position;
+        stats_line.time = buffer_line[time_begin_position .. time_end_position];
+
+        stats_line.line = buffer_line[time_end_position .. len];
+
+        return stats_line;
```

Comment:
So, in this method you always create new `StatsLine` and pass it to the caller. Creating `StatsLine` involves creating `Buffer` which `StatsLine` doesn't reuse in any way, and since it's private nobody else can do reuse it, so having `Buffer` there makes a little point (except to prevent copying StatsLines).

I think we should you could/should do it in a way where `StatsLine` is just a _view_ on the line.